aiogram/api/types/mixins.py

```diff
@@ -0,0 +1,89 @@
+import os
+import pathlib
+from typing import BinaryIO, Optional, Union
+
+from aiogram.api.types.file import File
+
+
+class Downloadable:
+    """
+    Mixin for files
+    """
+    async def download(
+        self,
+        destination: Optional[Union[BinaryIO, pathlib.Path, str]] = None,
+        timeout: int = 30,
+        chunk_size: int = 65536,
+        seek: bool = True,
+        make_dirs: bool = True,
+    ) -> Optional[BinaryIO]:
+        """
+        Download file
+
+        :param destination: Filename, file path or instance of :class:`io.IOBase`. For e.g. :class:`io.BytesIO`, defaults to None
+        :type destination: Optional[Union[BinaryIO, pathlib.Path, str]]
+        :param timeout: Total timeout in seconds, defaults to 30
+        :type timeout: int
+        :param chunk_size: File chunks size, defaults to 64 kb
+        :type chunk_size: int
+        :param seek: Go to start of file when downloading is finished. Used only for destination with :class:`typing.BinaryIO` type, defaults to True
+        :type seek: bool
+        :param make_dirs: make dirs if not exist
+        :return: destination
+        """
+        from aiogram.api.client.bot import Bot
+
+        file = await self.get_file()
+
+        is_path = True
+        if destination is None:
+            destination = file.file_path
+        elif isinstance(destination, (str, pathlib.Path)) and os.path.isdir(destination):
+            # file from self.get_file() always have file_path
```

Comment:
Why? Docs says File object has optional file_path: https://core.telegram.org/bots/api#file

Reply:
This is strange. File can only be obtained from the [getFile](https://core.telegram.org/bots/api#getfile) method, the description of which says "The file can then be downloaded via the link https://api.telegram.org/file/bot<token>/<file_path>, where **<file_path> is taken from the response**".

I thought that `file_path` would be missing for a large file (more than 20MB), but in this case an error is returned "Bad Request: file is too big". So I don't know when file_path can come empty. I asked a question to the technical support of bots.